### PR TITLE
Enhance unit tests for CodecService methods

### PR DESCRIPTION
**Pull request by ivamly**
# Enhance unit tests for CodecService methods

## Description:

This pull request improves the unit tests for the CodecService class by adding unit tests for the codec and availableCodecs methods. These enhancements aim to ensure robust coverage of these methods and verify their behavior. The coverage report in IntelliJ indicates that the line coverage for the CodecService class has increased from 86% to 90%.

## Related Issues:

#108706

## Checklist:

- [x]  I have signed the contributor license agreement.
- [x]  I have followed the contributor guidelines.
- [x]  I have built my formula locally prior to submission with gradle check.
- [x]  My PR is against the main branch. 
- [x]  I have checked that my submission is for an OS and architecture that we support.
- [x]  If I am submitting this code for a class, I have read our policy.

### CODE REVIEW

**Comment by ivamly** [comment]
Hi @astefan,

I’m new to open source, so I apologize if my questions or contributions come off as basic. I’m keen to learn and get better!

If you have any suggestions or need more changes for this pull request, just let me know. I’m happy to make any adjustments or address feedback you have.

Thanks for your time and help!

File: server/src/test/java/org/elasticsearch/index/codec/CodecTests.java [hunk 1]
```diff
@@ -96,6 +98,40 @@ public void testLegacyBestCompression() throws Exception {
...
+
+    public void testAvailableCodecsDoesNotContainUnknownCodec() throws Exception {
+        CodecService codecService = createCodecService();
+        String[] availableCodecs = codecService.availableCodecs();
+        List<String> codecList = Arrays.asList(availableCodecs);
+
+        assertFalse(codecList.contains("unknown_codec"));
+    }
+
+    public void testAvailableCodecsCount() throws Exception {
```
Comment (lkts) [inline server/src/test/java/org/elasticsearch/index/codec/CodecTests.java:127]: This is very similar to `testAvailableCodecsContainsExpectedCodecs`, i don't think it adds much value. Same for the test above.

**Review by lkts — COMMENTED** [review]
Thanks! What is the before/after code coverage?

**Comment by ivamly** [comment]
Thank you for the feedback! The coverage report in IntelliJ indicates that the line coverage for the `CodecService` class has increased from 86% to 90%.

File: server/src/test/java/org/elasticsearch/index/codec/CodecTests.java (hunk 1 above)
Comment (ivamly) [reply server/src/test/java/org/elasticsearch/index/codec/CodecTests.java:127]: I see that one solution could be to combine the checks into a single test to ensure that the expected codecs are present, that there are no unknown codecs, and to verify the total number of codecs. Should I update my PR to reflect this approach?

**Comment by lkts** [comment]
buildkite test this

File: server/src/test/java/org/elasticsearch/index/codec/CodecTests.java (hunk 1 above)
Comment (ivamly) [reply server/src/test/java/org/elasticsearch/index/codec/CodecTests.java]: I have made the requested changes and combined the checks into a single test. Is there anything else I can do to get the PR approved? Please let me know if there are any other modifications needed.
Thanks!

**Comment by lkts** [comment]
buildkite test this

**Comment by lkts** [comment]
@ivamly i don't know if you can see CI results but one of the failures is a style check. You need to do `./gradlew spotlessApply` to fix it and commit the results.

**Comment by lkts** [comment]
@elasticmachine update branch

**Comment by lkts** [comment]
buildkite test this

**Review by lkts — APPROVED** [review]
LGTM, thanks.

**Pull request merged**